src/univers/versions.py

```diff
@@ -385,6 +384,19 @@ class LegacyOpensslVersion(Version):
     univers.versions.InvalidVersion: '3.0.2' is not a valid <class 'univers.versions.LegacyOpensslVersion'>
     """
 
+    major = attr.ib(type=int, default=None, repr=False)
+    minor = attr.ib(type=int, default=None, repr=False)
+    build = attr.ib(type=int, default=None, repr=False)
+    patch = attr.ib(type=str, default=None, repr=False)
+
+    def __attrs_post_init__(self):
+        Version.__attrs_post_init__(self)
+        major, minor, build, patch = self.parse(self.string)
+        object.__setattr__(self, "major", major)
+        object.__setattr__(self, "minor", minor)
+        object.__setattr__(self, "build", build)
+        object.__setattr__(self, "patch", patch)
+
```

Comment:
I am not sure I like double parsing the string and overriding __attrs_post_init__ without calling super() too?
What about this code instead? (it is not 100% satisfying, but at least it does not double parse):
```
	
    @property
    def major(self):
        major, _minor, _build, _patch = self.value
        return major

    @property
    def minor(self):
        _major, minor, _build, _patch = self.value
        return minor

    @property
    def build(self):
        _major, _minor, build, _patch = self.value
        return build

    @property
    def patch(self):
        _major, _minor, _build, patch = self.value
        return patch
```
or at least reusing the self.value rather than reparsing it?

Reply:
LGTM :+1: